### PR TITLE
Add 'dataset' size to cat indices and cat shards

### LINKED ISSUE #95092: Add total_data_set_size_in_bytes to _cat/indices
### Description

`_cat/indices` does not currently show any size for searchable snapshots

`green  open   partial-.ds-mydatastream-2023.03.01-000579             7yUK80vZTHiqtWjX-q_hWQ   3   0     100047            0         0b             0b`
I couldn't find any extra columns that show this either.

The size of the actual backing index is available by querying stats:
`GET /partial-.ds-mydatastream-2023.03.01-000579/_stats`
```
      "store": {
        "size_in_bytes": 0,
        "total_data_set_size_in_bytes": 40627182,
        "reserved_in_bytes": 0
      },
```

It would be great if this was added as a column so that we can see the size of old indices (e.g. to see if they are increasing in size, how much space we could save if we delete them etc)

### PR DESCRIPTION
**Pull request by dakrone**
This adds the `dataset` computed size for the `/_cat/indices` and `/_cat/shards` APIs. This new column is reported by default.

Resolves #95092

### CODE REVIEW

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Review by joegallo — APPROVED** [review]
+1 to this change (as in, feel free to merge this PR).

Does it makes sense (as a separate PR, perhaps authored by somebody else) to document more / elsewhere that this new column is the right thing to use for searchable snapshot indices?

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -170,6 +170,7 @@ tasks.named("yamlRestTestV7CompatTransform").configure { task ->
   task.skipTest("service_accounts/10_basic/Test get service accounts", "new service accounts are added")
   task.skipTest("spatial/70_script_doc_values/diagonal length", "precision changed in 8.4.0")
   task.skipTest("spatial/70_script_doc_values/geoshape value", "error message changed in 8.9.0")
+  task.skipTest("security/authz/14_cat_indices/Test empty request while single authorized index", "not supported for compatibility")
```
Comment (joegallo) [inline x-pack/plugin/build.gradle:173]: Maybe replace `"not supported for compatibility"` with `"dataset.size added in 8.11.0"` or something like that? Could be a different PR, doesn't have to be this one. That is, if you're okay with that change but don't want to deal with another whole CI run on this PR, that's fine.

**Comment by dakrone** [comment]
> Does it makes sense (as a separate PR, perhaps authored by somebody else) to document more / elsewhere that this new column is the right thing to use for searchable snapshot indices?

I think we can do that when we finally document all of the cat indices options. I tried to make it clearer in the shards docs that this is for stuff like object storage.

**Pull request merged**